core/src/main/java/com/homeaway/streamplatform/streamregistry/db/dao/impl/ProducerDaoImpl.java

```diff
@@ -48,13 +51,21 @@
     private final List<String> topicPostFixes = Collections.singletonList("");
     private final static String ACTOR_TYPE = "producer";
 
+    @Inject
+    public ProducerDaoImpl(ManagedKafkaProducer managedKafkaProducer,
+        ManagedKStreams kStreams,
+        @Named("stream-registry-env") String env,
+        RegionDao regionDao,
+        ManagedInfraManager managedInfraManager) {
+        this(managedKafkaProducer, kStreams, env, regionDao, managedInfraManager.getInfraManager());
+    }
+
     public ProducerDaoImpl(ManagedKafkaProducer managedKafkaProducer,
         ManagedKStreams kStreams,
         String env,
         RegionDao regionDao,
-        InfraManager infraManager,
-        KafkaManager kafkaManager) {
-        super(managedKafkaProducer, kStreams, env, regionDao, infraManager, kafkaManager);
+        InfraManager infraManager) {
```

Comment:
If this constructor is not used.... delete it... and defer the `managedInfraManager.getInfraManager()` to internal field.

Reply:
Ideally we should pivot around `ManagedInfraManager` but seems existing IT impl leverages `InfraManager` and calls this constructor ( [here](https://github.com/homeaway/stream-registry/blob/master/core/src/test/java/com/homeaway/streamplatform/streamregistry/resource/BaseResourceIT.java#L248-L249) )

Reply:
Changed this flow.